lib/ThreeScaleClient.php

```diff
@@ -93,37 +91,42 @@ public function getHost() {
    *   ?>
    * </code>
    */
-  public function authorize($appId, $appKey = null, $serviceId = null, $usage = null) {
+  
+     public function authorize($appId, $appKey = null, $options, $usage = null) {
     $url = "http://" . $this->getHost() . "/transactions/authorize.xml";
-    $params = array('provider_key' => $this->getProviderKey(), 'app_id' => $appId);
+    $params = array('app_id' => $appId);
+
+    if ($options instanceof ThreeScaleClientCredentials ) {
+      $params['service_token'] = $options->service_token;
+      $params['service_id'] = $options->service_id;
+    } else {
+      $params['provider_key'] = $this->getProviderKey();
+      $params['service_id'] = $options;
+    }
 
     if ($appKey) {
       $params['app_key'] = $appKey;
     }
     
-    if ($serviceId) {
-      $params['service_id'] = $serviceId;
-    }
-
     if ($usage) {
       $params['usage'] = $usage;
     }
     
     $httpResponse = $this->httpClient->get($url, $params);
-
     if (self::isHttpSuccess($httpResponse)) {
       return $this->buildAuthorizeResponse($httpResponse->body);
     } else {
       return $this->processError($httpResponse);
     }
   }
-  
+
+
   /**
-   * Authorize an application.
+   * Authorize an application for OAuth auth mode.
    *
    * @param $appId  application id or client id (they are equivalent)
    * @param $usage usage
-   * @param $serviceId service id, required (from November 2016)
+   * @param $options accept service id or both credentials i.e (service id, service token)
```

Comment:
I **think** this could be using type annotation: https://phpdoc.org/docs/latest/references/phpdoc/tags/param.html

```
* @param ThreeScaleClientCredentials $options ...
* @param string $options ...
```